include/pm/policy.h

```diff
@@ -7,13 +7,46 @@
 #ifndef ZEPHYR_INCLUDE_PM_POLICY_H_
 #define ZEPHYR_INCLUDE_PM_POLICY_H_
 
+#include <stdbool.h>
 #include <stdint.h>
+
 #include <pm/state.h>
+#include <sys/slist.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
+/**
+ * @brief System Power Management Policy API
+ * @defgroup subsys_pm_sys_policy Policy
+ * @ingroup subsys_pm_sys
+ * @{
+ */
+
+/**
+ * @brief Callback to notify when when state allowed status changes.
+ *
+ * @param state State that changed
+ * @param allowed If the state is allowed or not.
+ */
+typedef void (*pm_policy_state_allowed_changed_cb_t)(enum pm_state state,
+						     bool allowed);
+
+/**
+ * @brief Callback to notify when maximum latency changes.
+ *
+ * @param latency New maximum latency in microseconds.
+ */
+typedef void (*pm_policy_latency_changed_cb_t)(int32_t latency);
```

Comment:
why latency is signed int? are negative values supported - if so what these mean?

Reply:
it is signed because we also need `SYS_FOREVER_US` (which is -1)

Reply:
let's add small info
```
Positive value represents latency in microseconds. Negative value of -1 lifts the latency constraint. Other values are forbidden.
```

Maybe `k_timeout_t` can be used here?

Reply:
clarified range of values. I'll check more in detail some existing opaque types and their usage (e.g. `k_ticks_t`, `k_timeout_t`, etc.).

Reply:
I would not bound this value to ticks, as there is a higre difference between 50ns and 50us of latency in some cases but all the values are well within one tick.

Reply:
> I would not bound this value to ticks, as there is a higre difference between 50ns and 50us of latency in some cases but all the values are well within one tick.

Not sure if I'm following, latency is specified in `us`, not ticks.

Reply:
Agree